lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -80,6 +80,49 @@ static Type lowerType(Type type) {
   return {};
 }
 
+/// This verifies that the target operation has been lowered to a legal
+// operation.  This checks that the operation recursively has no FIRRTL
+// operations or types.
+static LogicalResult verifyOpLegality(Operation *op) {
+  SmallVector<Operation *> worklist = {op};
+  while (!worklist.empty()) {
+    op = worklist.pop_back_val();
+
+    // Check that this operation is not a FIRRTL op.
+    if (op->getDialect() && (FIRRTLDialect::getDialectNamespace() ==
+                             op->getDialect()->getNamespace()))
```

Comment:
I'd recommend splitting this to the top of the function for readability:

`StringRef firrtlNamespace = FIRRTLDialect::getDialectNamespace();`

Or just use `op->getDialect()->getNamespace() == "firrtl"`

Reply:
Alternatively can you pointer compare `op->getDialect() == &FIRRTLDialect`?

Reply:
It looks like I can do `isa<FIRRTLDialect>(op->getDialect())` which compares TypeID instead. We might be able to do the following for a pointer compare, but I think i like the first thing better:
```c++
auto dialect = context->getLoadedDialect<FIRRTLDIalect>();
if (op->getDialect() == firrtlDialect)
```